lib/rucio/rse/protocols/webdav.py

```diff
@@ -184,12 +185,15 @@ def connect(self, credentials={}):
                 # Trying to get the proxy from the default location
                 proxy_path = '/tmp/x509up_u%s' % os.geteuid()
                 if os.path.isfile(proxy_path):
-                    x509 = proxy_path
+                    self.cert = (proxy_path, proxy_path)
                 elif self.auth_token:
+                    # If no proxy is found, we set the cert to None and use the auth_token
+                    self.cert = None
                     pass
                 else:
                     raise exception.RSEAccessDenied('X509_USER_PROXY is not set')
-            self.cert = (x509, x509)
+            else:
+                self.cert = (x509, x509)
```

Comment:
Could you separate this into a separate commit? It's a functional change that's unrelated to mutable params and type hints, and it's probably best to have it separately. You can reset the commit and `git add -p lib/rucio/rse/protocols/webdav.py` to only select these changes.

Reply:
Moved into a separate commit, 8e776054283ab2937c6ba7c10d25e10566a1101d